Makefile

```diff
@@ -81,25 +87,35 @@ all:
 	$(QUIET) make -C bpf/deserialization_to_bpf_map
 	
 	$(QUIET) $(GO) generate bpf/kmesh/bpf2go/bpf2go.go
-	
+
+	$(call printlog, BUILD, "kernel")
+	$(QUIET) make -C kernel/ko_src
+
+	$(QUIET) $(foreach app,$(BINARIES),$(call build_app,$(app));)
+
+build_kmesh-daemon:
 	$(call printlog, BUILD, $(APPS1))
 	$(QUIET) (export PKG_CONFIG_PATH=$(PKG_CONFIG_PATH):$(ROOT_DIR)mk; \
 		$(GO) build -ldflags $(LDFLAGS) -tags $(ENHANCED_KERNEL) -o $(APPS1) $(GOFLAGS) ./daemon/main.go)
-	
-	$(call printlog, BUILD, "kernel")
-	$(QUIET) make -C kernel/ko_src
 
+build_mdacore:
```

Comment:
nit: Add .PHONY. Can use `make build kmesh-cni` to compile the components directly.